src/argparse.zig

```diff
@@ -0,0 +1,297 @@
+const std = @import("std");
+
+const main = @import("main");
+const NeverFailingAllocator = main.heap.NeverFailingAllocator;
+const ListUnmanaged = main.ListUnmanaged;
+
+pub fn Parser(comptime T: type, comptime callback: ?fn(self: T) anyerror!void) type {
+	return struct {
+		const Self = @This();
+		pub const Args = T;
+
+		pub fn parse(allocator: NeverFailingAllocator, args: []const u8) !ParseResult(T) {
+			const result = _parse(allocator, args);
+			if(callback != null and result == .success) {
+				try callback.?(result.success);
+			}
+			return result;
+		}
+		pub fn _parse(allocator: NeverFailingAllocator, args: []const u8) ParseResult(T) {
```

Comment:
maybe you could find a better name for this?
Also this should probably be private.

Reply:
This most likely can be gone after callback is gone, it was used before because I don't like break to label syntax, I prefer early return, but now it is not really relevant I think

Reply:
Replaced with `resolve` that is used for parts of parse and autocomplete that are really similar.